langstream-api-gateway/src/main/java/ai/langstream/apigateway/gateways/TopicProducerCacheFactory.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.langstream.apigateway.gateways;
+
+import ai.langstream.apigateway.config.TopicProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class TopicProducerCacheFactory {
+
+    @Bean
+    public TopicProducerCache topicProducerCache(TopicProperties topicProperties) {
+        if (topicProperties.isProducersCacheEnabled()) {
+            return new LRUTopicProducerCache(topicProperties.getProducersCacheSize());
+        } else {
+            return (key, topicProducerSupplier) -> topicProducerSupplier.get();
```

Comment:
This is tricker then expected, see the comments above
the producer that is returned should be already ready to accept writes (started)

Reply:
yes it is